### PR TITLE
Set LC_ALL=C before running ping process in Ping.Send()

### LINKED ISSUE #50363: Ping.Send() throws PingException on Linux non-English locale
### Description

On non-EN locale, Ping.Send() throws PingException.
It seem to fail parsing localized output of `ping` command.

Repro code:

```cs
using System;
using System.Net.NetworkInformation;

using var ping = new Ping();

var reply = ping.Send("192.168.0.1");

Console.WriteLine($"{reply.Status} {reply.Address}, {reply.RoundtripTime}[ms]");
```

If you change the `LANG` environment variable, the runtime will output different results.

```
$ echo $LANG
ja_JP.UTF-8

$ dotnet run
Unhandled exception. System.Net.NetworkInformation.PingException: An exception occurred during a Ping request.
 ---> System.Net.NetworkInformation.PingException: An exception occurred during a Ping request.
   at System.Net.NetworkInformation.Ping.SendWithPingUtility(IPAddress address, Byte[] buffer, Int32 timeout, PingOptions options)
   at System.Net.NetworkInformation.Ping.SendPingCore(IPAddress address, Byte[] buffer, Int32 timeout, PingOptions options)
   at System.Net.NetworkInformation.Ping.Send(IPAddress address, Int32 timeout, Byte[] buffer, PingOptions options)
   --- End of inner exception stack trace ---
   at System.Net.NetworkInformation.Ping.Send(IPAddress address, Int32 timeout, Byte[] buffer, PingOptions options)
   at System.Net.NetworkInformation.Ping.Send(String hostNameOrAddress, Int32 timeout, Byte[] buffer, PingOptions options)
   at System.Net.NetworkInformation.Ping.Send(String hostNameOrAddress)
   at <Program>$.<Main>$(String[] args) in test.cs:line 6

$ LANG=C dotnet run
Success 192.168.0.1, 0[ms]

$ LANG=en-US.UTF-8 dotnet run
Success 192.168.0.1, 0[ms]
```

On `LANG=C`, gets expected result.
Also on `LANG=en-US`, gets expected result.
However, as an example on `LANG=ja_JP`, will cause PingException as above.

Outputs of `ping` command on each locales are like below.

```
$ echo $LANG
ja_JP.UTF-8

$ ping 192.168.0.1
PING 192.168.0.1 (192.168.0.1) 56(84) バイトのデータ
64 バイト応答 送信元 192.168.0.1: icmp_seq=1 ttl=64 時間=0.359ミリ秒
64 バイト応答 送信元 192.168.0.1: icmp_seq=2 ttl=64 時間=0.320ミリ秒
^C
--- 192.168.0.1 ping 統計 ---
送信パケット数 2, 受信パケット数 2, パケット損失 0%, 時間 1022ミリ秒
rtt 最小/平均/最大/mdev = 0.320/0.339/0.359/0.019ミリ秒

$ LANG=C ping 192.168.0.1
PING 192.168.0.1 (192.168.0.1) 56(84) bytes of data.
64 bytes from 192.168.0.1: icmp_seq=1 ttl=64 time=0.428 ms
64 bytes from 192.168.0.1: icmp_seq=2 ttl=64 time=0.393 ms
^C
--- 192.168.0.1 ping statistics ---
2 packets transmitted, 2 received, 0% packet loss, time 1029ms
rtt min/avg/max/mdev = 0.393/0.410/0.428/0.017 ms
```

### Configuration
- .NET SDK 5.0.201, 6.0.0-preview.2.21154.6
- Ubuntu 20.04.2 LTS

### Other information
I think this issue may be fixed by adding `ProcessStartInfo.EnvironmentVariables.Add("LANG", "C")` before running `ping` command.

### PR DESCRIPTION
**Pull request by smdn**
Fix #50363

### CODE REVIEW

**Review by scalablecory — APPROVED** [review]
Looks reasonable to me.

CC @wfurt for final approval

**Review by wfurt — APPROVED** [review]
LGTM
Thanks for the fix @smdn

File: src/libraries/System.Net.Ping/src/System.Net.Ping.csproj [hunk 1]
```diff
@@ -102,6 +102,7 @@
     <Reference Include="System.Threading.ThreadPool" />
   </ItemGroup>
   <ItemGroup Condition="'$(TargetsUnix)' == 'true'">
+    <Reference Include="System.Collections.Specialized" />
```
Comment (danmoseley) [inline src/libraries/System.Net.Ping/src/System.Net.Ping.csproj:105]: Is this needed?

Comment (wfurt) [reply src/libraries/System.Net.Ping/src/System.Net.Ping.csproj:105]: seems like. I had same doubt and build fails without it.

**Comment by smdn** [comment]
Thank you for reviewing and merging!

**Pull request merged**